app/tests/challenges_tests/test_models.py

```diff
@@ -120,18 +121,31 @@ def test_total_challenge_cost():
     request3 = ChallengeRequestFactory(
         creator=normal_user, expected_number_of_teams=10
     )
+    request4 = ChallengeRequestFactory(
+        creator=normal_user,
+        expected_number_of_teams=10,
+        algorithm_selectable_gpu_type_choices=["", "A10G", "T4"],
+    )
 
     organisation = OrganizationFactory(exempt_from_base_costs=True)
     organisation.members_group.user_set.add(user_exempt_from_base_cost)
 
-    assert request1.storage_and_compute_cost_surplus == -300
-    assert request1.total_challenge_cost == 1000
+    original_value = settings.COMPONENTS_DEFAULT_BACKEND
+    settings.COMPONENTS_DEFAULT_BACKEND = "grandchallenge.components.backends.amazon_sagemaker_training.AmazonSageMakerTrainingExecutor"
+    try:
+        assert request1.storage_and_compute_cost_surplus == -270
+        assert request1.total_challenge_cost == 1000
+
+        assert request2.storage_and_compute_cost_surplus == -270
+        assert request2.total_challenge_cost == 6000
 
-    assert request2.storage_and_compute_cost_surplus == -300
-    assert request2.total_challenge_cost == 6000
+        assert request3.storage_and_compute_cost_surplus == 1380
+        assert request3.total_challenge_cost == 7500
 
-    assert request3.storage_and_compute_cost_surplus == 1290
-    assert request3.total_challenge_cost == 7500
+        assert request4.storage_and_compute_cost_surplus == 2580
+        assert request4.total_challenge_cost == 9000
+    finally:
+        settings.COMPONENTS_DEFAULT_BACKEND = original_value
```

Comment:
You do not need to keep the original value and then set it back, the fixture handles rollback automatically. The `try`/`finally` can also go.

Reply:
Yes, I was a little confused by this. It should not be necessary, but without it I get a lot of failures in the tests. The current job is failing again.

Reply:
I see you marked this as resolved. I had reverted these changes. Do you want me to bring them back again?

Reply:
Sorry, you had only pushed the one commit removing the try finally and had closed the other conversation without pushing anything, I thought you had clicked the wrong one. The test failures are strange, that shouldn't happen.

Reply:
It is because `from django.conf import settings` is being used in the test. This should never be used in tests, the fixture should be used instead.

Reply:
I have pushed a fix, the test still fails with:

```
>       assert request1.storage_and_compute_cost_surplus == -270
E       assert 410 == -270
```

I am not sure what the correct thing to do there is. But now you shouldn't get unrelated errors in other tests.

Reply:
It appears to be caused by db9db17, which seems odd.

Reply:
> It is because `from django.conf import settings` is being used in the test. This should never be used in tests, the fixture should be used instead.

*Slaps forehead*
Thank you!

Reply:
Ah, it is because the challenge request factory was modified.

Reply:
> I have pushed a fix, the test still fails with:
> 
> ```
> >       assert request1.storage_and_compute_cost_surplus == -270
> E       assert 410 == -270
> ```
> 
> I am not sure what the correct thing to do there is. But now you shouldn't get unrelated errors in other tests.

That test failure was related to a later change. Just fixed it.